spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java

```diff
@@ -588,10 +589,12 @@ public JSONArray getTargetedSequencingTypes(Long libraryPrepKitId) throws IOExce
       Map<String, Object> targetedSequencingMap = Maps.newHashMap();
       targetedSequencingMap.put("targetedSequencingId", targetedSequencing.getId());
       targetedSequencingMap.put("alias", targetedSequencing.getAlias());
-      targetedSequencingMap.put("kitDescriptorId", targetedSequencing.getKitDescriptor().getId());
-      fullTargetedSequencingCollection.add(targetedSequencingMap);
-      if (libraryPrepKitId != null && libraryPrepKitId.equals(targetedSequencing.getKitDescriptor().getId())) {
-        targetedSequencingByKit.add(targetedSequencingMap);
+      for (KitDescriptor kitDescriptor : targetedSequencing.getKitDescriptors()) {
+        targetedSequencingMap.put("kitDescriptorId", kitDescriptor.getId());
+        fullTargetedSequencingCollection.add(targetedSequencingMap);
+        if (libraryPrepKitId != null && libraryPrepKitId.equals(kitDescriptor.getId())) {
+          targetedSequencingByKit.add(targetedSequencingMap);
+        }
```

Comment:
This method can be greatly simplified - just fetch the kitDescriptor by id and get its targetedSequencings instead of all this